src/markdownEngine.ts

```diff
@@ -2,70 +2,71 @@
 
 import { extensions, workspace, WorkspaceConfiguration } from 'vscode';
 import { slugify } from './util';
+import { MarkdownContribution, MarkdownContributionProvider, getMarkdownContributionProvider } from './markdownExtensions'
 
 class MarkdownEngine {
-    public md?;
+    public cacheMd;
 
+    public async getEngine() {
+        if (!this.cacheMd) {
+            this.cacheMd = await this.newEngine();
+        }
+        return this.cacheMd;
+    }
+
+    public contributionsProvider = getMarkdownContributionProvider();
     private _slugCount = new Map<string, number>();
 
     constructor() {
-        this.initMdIt();
+        this.contributionsProvider.onContributionsChanged(() => {
+            this.cacheMd = null;
+            this.getEngine();
+        });
+        this.cacheMd = null;
+        this.getEngine();
     }
 
-    private async initMdIt() {
-        if (!this.md) {
-            const hljs = await import('highlight.js');
-            const mdtl = await import('markdown-it-task-lists');
-            const mdkt = await import('@neilsustc/markdown-it-katex');
-
-            //// Make a deep copy as `macros` will be modified by KaTeX during initialization
-            let userMacros = JSON.parse(JSON.stringify(workspace.getConfiguration('markdown.extension.katex').get<object>('macros')));
-            let katexOptions = { throwOnError: false };
-            if (Object.keys(userMacros).length !== 0) {
-                katexOptions['macros'] = userMacros;
-            }
+    private async newEngine() {
+        var md;
+
+        const hljs = await import('highlight.js');
 
-            this.md = (await import('markdown-it'))({
-                html: true,
-                highlight: (str: string, lang?: string) => {
-                    lang = normalizeHighlightLang(lang);
-                    if (lang && hljs.getLanguage(lang)) {
-                        try {
-                            return `<div>${hljs.highlight(lang, str, true).value}</div>`;
-                        }
-                        catch (error) { }
+        md = (await import('markdown-it'))({
+            html: true,
+            highlight: (str: string, lang?: string) => {
+                lang = normalizeHighlightLang(lang);
+                if (lang && hljs.getLanguage(lang)) {
+                    try {
+                        return `<div>${hljs.highlight(lang, str, true).value}</div>`;
                     }
-                    return `<code><div>${this.md.utils.escapeHtml(str)}</div></code>`;
+                    catch (error) { }
                 }
-            }).use(mdtl).use(mdkt, katexOptions);
```

Comment:
Looks like you forgot to add them back.

Reply:
No, it will be processed in `extendMarkdownIt` of this extension's. All the `extendMarkdownIt` function will be called, including this extension's.

Reply:
👍